### PR TITLE
Custom eslint report formatter

### PR DESCRIPTION
**Pull request by mayank23**
- new report formatter better suited for this plugin. Can handle error message of varying length.
- more colors 💯 
- updates the the readme 
- might decide in v3 to create a separate package for the formatter
invalid-icu-syntax-screenshot

### CODE REVIEW

File: formatter.js [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+const path = require('path');
+
+const CWD = process.cwd();
+
+const formatter = (results) => {
+  const total = {
+    errors: 0,
+    warnings: 0
+  };
+  const outputs = results.map(({ filePath, messages , errorCount, warningCount}) => {
```
Comment (silasbw) [inline formatter.js]: Did you run linters over this file?  There's a few style issues I'd expect a linter to pick up.

Run the linters :)

File: examples/multiple-files-per-locale/translations/hi-IN/todos.json [hunk 2]
```diff
@@ -0,0 +1,9 @@
+{
+  "todos.addTodo": "नया टोडो जोड़ें",
+  "todos.count": "आपके पास {count, plural, =0 {कोई टॉडोस नहीं है} other {# टोडो है} }",
+  "todos.greeting": "स्वागत हे {userName}",
+  "todos.markCompleted": "पूर्ण के रूप में चिह्नित करें",
+  "todos.nextPage": "आगामी",
+  "todos.previousPage": "पिछला",
+  "todos.removeTodo": "टूडू को मिटाएं"
+}
```
Comment (silasbw) [inline examples/multiple-files-per-locale/translations/hi-IN/todos.json]: aafn

File: examples/multiple-files-per-locale/translations/hi-IN/search-results.json [hunk 3]
```diff
@@ -0,0 +1,4 @@
+{
+  "search.noResults": "सभी नहीं मिला!",
+  "search.placeholder": "सभी को ढूंढें"
+}
```
Comment (silasbw) [inline examples/multiple-files-per-locale/translations/hi-IN/search-results.json]: affn

File: examples/multiple-files-per-locale/translations/hi-IN/login.json [hunk 4]
```diff
@@ -0,0 +1,11 @@
...
+  "login": {
+    "form": {
+      "cancel": "रद्द करना",
+      "resetPassword": "पासवर्ड रीसेट",
+      "submit": "अब प्रवेश करें!"
+    },
+    "invalidAuth": "गलत उपयोगकर्ता नाम या पासवर्ड",
+    "lastLoginAt": "पर अंतिम लॉगिन {lastLoginTime, time, short}"
+  }
+}
```
Comment (silasbw) [inline examples/multiple-files-per-locale/translations/hi-IN/login.json]: aafn

File: examples/multiple-files-per-locale/package.json [hunk 5]
```diff
@@ -2,15 +2,15 @@
...
   "license": "MIT",
   "main": "index.js",
   "name": "multiple-files-per-locale",
   "scripts": {
     "test": "echo \"Error: no test specified\" && exit 1",
-    "lint": "eslint --fix --ext .json translations"
+    "lint": "eslint --fix --ext .json  --format node_modules/eslint-plugin-i18n-json/formatter.js translations/"
   },
   "version": "1.0.0"
 }
```
Comment (silasbw) [inline examples/multiple-files-per-locale/package.json]: aafn

File: README.md [hunk 6]
```diff
@@ -96,15 +103,41 @@ simple
...
-    <a align="center" href="https://asciinema.org/a/jIu2dJVuVucawqaFmgjuSUPg0" target="_blank"><img src="https://asciinema.org/a/jIu2dJVuVucawqaFmgjuSUPg0.png" /></a>
+    ![](assets/invalid-icu-syntax-screenshot.png)
 
 ## Examples
 Check out the [Examples](examples/) folder to see different use cases.
 
 ## Configuring the rules
 - Simply update your `.eslintrc.*` with overrides for the individual rules.
 - Eslint severities: 2 = error, 1 = warning, 0 = off
+- Example of the module's default rule configuration:
```
Comment (silasbw) [inline README.md]: Are the example-related changes part of the formatter work?  If not, consider pulling them into a separate PR.

File: formatter.js (hunk 1 above)
Comment (mayank23) [reply formatter.js]: thanks, must have left this out from being linted

File: README.md (hunk 6 above)
Comment (mayank23) [reply README.md]: yes, for the most part they are. (although, I did extend an example)

**Comment by mayank23** [comment]
@silasbw updated the PR 😄

File: formatter.js [hunk 7]
```diff
@@ -0,0 +1,54 @@
...
+const chalk = require('chalk');
+const plur = require('plur');
+const logSymbols = require('log-symbols');
+const indentString = require('indent-string');
+const path = require('path');
+
+const CWD = process.cwd();
+
+const formatter = (results) => {
+  const total = {
```
Comment (gergelyke) [inline formatter.js]: can we find a better name for it (that describes what totals are we talking about)?

File: formatter.js [hunk 8]
```diff
@@ -0,0 +1,54 @@
...
+    errors: 0,
+    warnings: 0,
+  };
+  const outputs = results.map(({
```
Comment (gergelyke) [inline formatter.js]: what are `outputs`? should we name it `errorLines`, or something more descriptive?

File: examples/README.md [hunk 9]
```diff
@@ -0,0 +1,9 @@
+# Project Use Examples
+
+- All examples are designed so it's easy to get started and see how the plugin can be configured to fit your needs.
+
+- **Thanks for checking out the examples! 😀**
+
+## Disclaimer
+
+- *None of the translations in the examples reflect actual GoDaddy translations for the various languages used.* They were just created using Google Translate for example sake 😉.
```
Comment (gshively11) [inline examples/README.md]: grammar typo: `for example's sake` or `for the sake of example`

File: README.md [hunk 10]
```diff
@@ -68,8 +72,8 @@ simple
 ```
 
 **In this project directory, do the following:**
-1) >npm install --save-dev eslint-plugin-i18n-json 
-2) Create a `.eslintrc.js` file in the root dir of your project.
+1) >npm install --save-dev eslint-plugin-i18n-json
```
Comment (silasbw) [inline README.md]: is the block quote intentional?  do you want fixed width instead?

File: formatter.js [hunk 11]
```diff
@@ -0,0 +1,54 @@
...
+    }
+    total.errors += errorCount;
+    total.warnings += warningCount;
+    const relativePath = path.relative(CWD, filePath);
+    const header = `${chalk.underline.white(relativePath)}`;
+
+    messages.sort((a, b) => b.severity - a.severity);
+
+    const messagesOutput = messages.map(({ ruleId, severity, message }) => {
+      let messageHeader = `${severity === 1 ? `${logSymbols.warning} ${chalk.inverse.yellow(' WARNING ')}` : `${logSymbols.error} ${chalk.inverse.red(' ERROR ')}`}`;
```
Comment (silasbw) [inline formatter.js]: I'm not sure the template expressions help readability here.  How about:

```js
let header = severity === 1 ? `${logSymbols.warning} ${chalk.inverse.yellow(' WARNING ')}` : `${logSymbols.error} ${chalk.inverse.red(' ERROR ')}`
```

File: formatter.js [hunk 12]
```diff
@@ -0,0 +1,54 @@
...
+      messageHeader += ` ${chalk.white(`(${ruleId})`)}`;
```
Comment (silasbw) [inline formatter.js]: does this evaluation to the same thing?  (the nested template expressions are hard to parse)

```js
messageHeader += ' ' + chalk.white(`(${ruleId})`);
```

File: formatter.js (hunk 8 above)
Comment (mayank23) [reply formatter.js]: agreed, will make it more descriptive.

File: README.md (hunk 10 above)
Comment (mayank23) [reply README.md]: yes, it's intentional, I wanted to make terminal commands have a distinct look

File: formatter.js (hunk 12 above)
Comment (mayank23) [reply formatter.js]: agreed, will update. originally had it that way, but was adhering to https://eslint.org/docs/rules/no-useless-concat
